pkg/cpio/newc/newc.go

```diff
@@ -117,6 +117,9 @@ func (w *writer) WriteRecord(f cpio.Record) error {
 
 	buf := &bytes.Buffer{}
 	hdr := headerFromInfo(f.Info)
+	if f.ReadCloser == nil {
```

Comment:
So what does GNU cpio do? Does it set file size to 0 for directories? Or does it ignore file size of directories when recreating files? Ideally our cpio-generated archives should be compatible.